### ISSUE TITLE
can't replay on mounted volume.

### ISSUE THREAD

**Issue by Vasiliy-Bondarenko** [EOSIO/eos#5685]
my state dir is actually a mount point.
the problem is when i try to make --hard-replay nodeos is trying to kill this directory. and can't. and it stops there.
```
2018-09-16T14:39:01.889 thread-0   chain_plugin.cpp:315          plugin_initialize    ] initializing chain plugin
2018-09-16T14:39:01.890 thread-0   chain_plugin.cpp:479          plugin_initialize    ] Replay requested: deleting state database
2018-09-16T14:39:01.895 thread-0   chain_plugin.cpp:629          plugin_initialize    ] 13 N5boost10filesystem16filesystem_errorE: boost::filesystem::remove: Device or resource busy: "/home/parallels/eos/JungleTestnet-eosbpstatsco/state"
rethrow boost::filesystem::remove: Device or resource busy: "/home/parallels/eos/JungleTestnet-eosbpstatsco/state": 
    {"what":"boost::filesystem::remove: Device or resource busy: \"/home/parallels/eos/JungleTestnet-eosbpstatsco/state\""}
    thread-0  chain_plugin.cpp:629 plugin_initialize
2018-09-16T14:39:01.896 thread-0   main.cpp:125                  main                 ] 13 N5boost10filesystem16filesystem_errorE: boost::filesystem::remove: Device or resource busy: "/home/parallels/eos/JungleTestnet-eosbpstatsco/state"
rethrow boost::filesystem::remove: Device or resource busy: "/home/parallels/eos/JungleTestnet-eosbpstatsco/state": 
    {"what":"boost::filesystem::remove: Device or resource busy: \"/home/parallels/eos/JungleTestnet-eosbpstatsco/state\""}
    thread-0  chain_plugin.cpp:629 plugin_initialize
```
```
> df -h
Filesystem      Size  Used Avail Use% Mounted on
/dev/sda1        62G   55G  4.4G  93% /
... [spikked] ...
/dev/sdb         40G   48M   38G   1% /home/parallels/eos/JungleTestnet-eosbpstatsco/state
```

**Comment by taokayan** [comment]
Did you have multiple eos instance accessing the same file? also in some linux versions you need to use "sudo" in order to get the file mounted to the memory.

**Comment by Vasiliy-Bondarenko** [comment]
@taokayan, nodeos can not delete moint point. This is a problem. What it should do instead is delete everything inside.

**Comment by arhag** [comment]
I think it may be worth changing the behavior of replay to only delete the contents inside of the `state` folder. However, there is a workaround you can already use. Instead of mounting the `state` folder, mount the parent `data` the same way you would have mounted the `state` folder. If you don't want `blocks` to be stored in that volume, then you can override its path using the `blocks-dir` config option (make sure to provide an absolute path).

**Comment by arhag** [comment]
Resolved by #5696.

**Comment by Vasiliy-Bondarenko** [comment]
@arhag Thanx.
